### PR TITLE
Migrate JAX ResNet50 benchmarks

### PR DESCRIPTION
**Pull request by pzread**
Migrate JAX ResNet50 models from https://github.com/iree-org/iree-samples/tree/main/oobi/benchmark-definitions/python

Currently we are only benchmarking FP32 models. So for the model and test data artifacts I only moved FP32 version. The rest are left as TODO to reduce conflicts with #12

### CODE REVIEW

File: common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py [hunk 1]
```diff
@@ -217,3 +217,159 @@
...
+    id=utils.BATCH_ID(unique_ids.OUTPUT_DATA_RESNET50_FP32_JAX_2048X7X7XF32),
+    name=utils.BATCH_NAME("OUTPUT_DATA_RESNET50_FP32_JAX_2048X7X7XF32"),
+    tags=["input-data", "imagenet", utils.BATCH_TAG],
+    source_info="",
+    artifacts={
+        def_types.ModelTestDataFormat.NUMPY_TENSORS:
+            utils.ModelTestDataArtifactTemplate(
+                data_format=def_types.ModelTestDataFormat.NUMPY_TENSORS,
+                data_parameters={
+                    "tensor_names": ["pixel_values"],
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py:309]:
```suggestion
                    "tensor_names": ["output_0"],
```

File: common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py [hunk 2]
```diff
@@ -217,3 +217,159 @@
...
+    id=utils.BATCH_ID(unique_ids.OUTPUT_DATA_RESNET50_FP16_JAX_2048X7X7XF16),
+    name=utils.BATCH_NAME("OUTPUT_DATA_RESNET50_FP16_JAX_2048X7X7XF16"),
+    tags=["input-data", "imagenet", utils.BATCH_TAG],
+    source_info="",
+    artifacts={
+        def_types.ModelTestDataFormat.NUMPY_TENSORS:
+            utils.ModelTestDataArtifactTemplate(
+                data_format=def_types.ModelTestDataFormat.NUMPY_TENSORS,
+                data_parameters={
+                    "tensor_names": ["pixel_values"],
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py:331]:
```suggestion
                    "tensor_names": ["output_0"],
```

File: common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py [hunk 3]
```diff
@@ -217,3 +217,159 @@
...
+    id=utils.BATCH_ID(unique_ids.OUTPUT_DATA_RESNET50_BF16_JAX_2048X7X7XBF16),
+    name=utils.BATCH_NAME("OUTPUT_DATA_RESNET50_BF16_JAX_2048X7X7XBF16"),
+    tags=["input-data", "imagenet", utils.BATCH_TAG],
+    source_info="",
+    artifacts={
+        def_types.ModelTestDataFormat.NUMPY_TENSORS:
+            utils.ModelTestDataArtifactTemplate(
+                data_format=def_types.ModelTestDataFormat.NUMPY_TENSORS,
+                data_parameters={
+                    "tensor_names": ["pixel_values"],
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py:353]:
```suggestion
                    "tensor_names": ["output_0"],
```

File: common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    image_processor = AutoImageProcessor.from_pretrained(MODEL_NAME)
+    inputs = image_processor(images=image, return_tensors="jax")
+    tensor = inputs["pixel_values"]
+    tensor = jnp.asarray(jnp.tile(tensor, [self.batch_size, 1, 1, 1]),
+                         dtype=self.model.dtype)
+    return (tensor,)
+
+  def forward(self, inputs: Tuple[Any, ...]) -> Tuple[Any, ...]:
+    tensor, = inputs
+    output = self.model(tensor)[0]
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py:69]:
```suggestion
    output = self.model(tensor).last_hidden_state
```

File: common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  """
+  # We use an image of 5 applies since this is an easy example.
+  img_path = "https://storage.googleapis.com/iree-model-artifacts/ILSVRC2012_val_00000023.JPEG"
+  data = requests.get(img_path).content
+  img = Image.open(io.BytesIO(data))
+  img = img.resize((width, height))
+  return img
+
+
+class ResNet50(model_interfaces.InferenceModel):
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py:32]: ResNets can come in different variants e.g. ResNet101. Similar to Bert and T5, let's rename this class to ResNet and include the `model_name` as a parameter, where it defaults to "microsoft/resnet-50".

File: common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+DTYPE_MAP = {
+    "fp32": jnp.float32,
+    "fp16": jnp.float16,
+    "bf16": jnp.bfloat16,
+}
+
+
+def create_model(batch_size: int = 1,
+                 data_type: str = "fp32",
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py:85]: Add model_name: str = "microsoft/resnet-50" as a param.

File: common_benchmark_suite/openxla/benchmark/unique_ids.py [hunk 7]
```diff
@@ -56,6 +56,34 @@
 OUTPUT_DATA_BERT_LARGE_FP16_JAX_384X1024XF16 = f"{OUTPUT_DATA_BERT_LARGE_JAX_384X1024}-fp16"
 OUTPUT_DATA_BERT_LARGE_BF16_JAX_384X1024XBF16 = f"{OUTPUT_DATA_BERT_LARGE_JAX_384X1024}-bf16"
 
+################################################################################
+# ResNet50 models                                                              #
+################################################################################
+MODEL_IMPL_RESNET50 = f"aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_IMPL"
+MODEL_RESNET50 = "aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_RESNET50"
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/unique_ids.py:63]: Might need a `MODEL_RESNET` and then a `MODEL_RESNET50`, as `MODEL_BERT` is to `MODEL_BERT_LARGE`.

File: common_benchmark_suite/openxla/benchmark/unique_ids.py [hunk 8]
```diff
@@ -56,6 +56,34 @@
 OUTPUT_DATA_BERT_LARGE_FP16_JAX_384X1024XF16 = f"{OUTPUT_DATA_BERT_LARGE_JAX_384X1024}-fp16"
 OUTPUT_DATA_BERT_LARGE_BF16_JAX_384X1024XBF16 = f"{OUTPUT_DATA_BERT_LARGE_JAX_384X1024}-bf16"
 
+################################################################################
+# ResNet50 models                                                              #
+################################################################################
+MODEL_IMPL_RESNET50 = f"aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_IMPL"
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/unique_ids.py:62]:
```suggestion
MODEL_IMPL_RESNET = f"aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_IMPL"
```

File: common_benchmark_suite/openxla/benchmark/unique_ids.py [hunk 9]
```diff
@@ -56,6 +56,34 @@
...
+MODEL_RESNET50_FP32_JAX = f"{MODEL_RESNET50_FP32}-JAX"
+MODEL_RESNET50_FP32_JAX_3X224X224XF32 = f"{MODEL_RESNET50_FP32_JAX}-3x224x224xf32"
+
+MODEL_RESNET50_FP16_JAX = f"{MODEL_RESNET50_FP16}-JAX"
+MODEL_RESNET50_FP16_JAX_3X224X224XF16 = f"{MODEL_RESNET50_FP16_JAX}-3x224x224xf16"
+
+MODEL_RESNET50_BF16_JAX = f"{MODEL_RESNET50_BF16}-JAX"
+MODEL_RESNET50_BF16_JAX_3X224X224XBF16 = f"{MODEL_RESNET50_BF16_JAX}-3x224x224xbf16"
+
+INPUT_DATA_RESNET50_FP32_JAX_3X224X224XF32 = "b2676fcd-1cc1-4b79-84a2-71c7efaeb58b"
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/unique_ids.py:80]: Should these have the `-fp32`, `-fp16` and `-bf16` suffixes?

Comment (pzread) [reply common_benchmark_suite/openxla/benchmark/unique_ids.py]: Done.

File: common_benchmark_suite/openxla/benchmark/unique_ids.py (hunk 7 above)
Comment (pzread) [reply common_benchmark_suite/openxla/benchmark/unique_ids.py:63]: Done.

File: common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py (hunk 6 above)
Comment (pzread) [reply common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py]: Done.

File: common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py (hunk 5 above)
Comment (pzread) [reply common_benchmark_suite/openxla/benchmark/models/jax/resnet50/resnet50_model.py]: Done.

File: .github/workflows/run_comparative_benchmark.yml [hunk 10]
```diff
@@ -13,6 +13,7 @@ on:
     # Scheduled to run at 09:00 UTC and 21:00 UTC.
     - cron: '0 09,21 * * *'
   workflow_dispatch:
+  pull_request:
```
Comment (pzread) [inline .github/workflows/run_comparative_benchmark.yml:16]: Reminder of removal

File: common_benchmark_suite/openxla/benchmark/unique_ids.py [hunk 11]
```diff
@@ -56,6 +56,34 @@
...
+################################################################################
+# ResNet models                                                                #
+################################################################################
+MODEL_IMPL_RESNET = "aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_IMPL"
+MODEL_RESNET50 = "aff75509-4420-40a8-844e-dbfc48494fe6-MODEL_RESNET50"
+MODEL_RESNET50_FP32 = f"{MODEL_RESNET50}-fp32"
+MODEL_RESNET50_FP16 = f"{MODEL_RESNET50}-fp16"
+MODEL_RESNET50_BF16 = f"{MODEL_RESNET50}-bf16"
+
+# JAX ResNet mdoels
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/unique_ids.py:68]:
```suggestion
# JAX ResNet models.
```

**Review by mariecwhite — APPROVED** [review]

**Pull request merged**